flyteplugins/go/tasks/pluginmachinery/ioutils/remote_file_output_reader.go

```diff
@@ -123,9 +123,13 @@ func (r RemoteFileOutputReader) DeckExists(ctx context.Context) (bool, error) {
 }
 
 func NewRemoteFileOutputReader(_ context.Context, store storage.ComposedProtobufStore, outPaths io.OutputFilePaths, maxDatasetSize int64) RemoteFileOutputReader {
+	maxPayloadSize := maxDatasetSize
+	if maxPayloadSize == 0 {
+		maxPayloadSize = storage.GetConfig().Limits.GetLimitMegabytes * 1024 * 1024
+	}
 	return RemoteFileOutputReader{
 		outPath:        outPaths,
 		store:          store,
-		maxPayloadSize: maxDatasetSize,
+		maxPayloadSize: maxPayloadSize,
```

Comment:
is there an issue with just removing the `maxPayloadSize` completely? IIUC there is just the two checks in [this file](github.com/flyteorg/flyte/blob/master/flyteplugins/go/tasks/pluginmachinery/ioutils/remote_file_output_reader.go). IMO the `maxDownloadMBs` applied on the data store retrieval would catch these first.